recipes/psi4/meta.yaml

```diff
@@ -0,0 +1,212 @@
+# * so the `{{ mkl }}  # [win]` below pins to 2022.1.0 so that libblas=*=*mkl can work correctly to set the blas implementation.
+# * without the `{{ mkl }}` pin, various attempts would either install mkl + openblas + libblas=*=*openblas or try to fall back to libblas==3.8.0=10_mkl
+# * current conda-forge-pinning is mkl=mkl-devel=2022, so hopefully this recipe is still consistent with the c-f stack.
+
+# bkgd for review: the windows openmp story
+# * the windows mkl package depends on intel-openmp (provides libiomp5md.dll).
+# * llvm-openmp is another contender on windows (provides libomp.dll=libiomp5md.dll and assoc. import libs libomp.lib=libiomp5md.lib).
+# * intel-openmp libiomp5md.dll is twice the size of the llvm-openmp file of the same name, and there's reports of non-interchangeability:
+#   * https://github.com/conda-forge/conda-forge.github.io/issues/1597#issuecomment-1029224521
+#   * https://github.com/conda-forge/intel_repack-feedstock/issues/1#issuecomment-579975070
+# * so we'd like to keep depending on intel-openmp, but since the package only provides the runtime dll,
+#   not the buildtime import lib, we stash a copy of the latter in a repo (see 2nd source/url above)
+#   for compiling psi4. It is used at buildtime and never packaged.
+# * for threading on windows, we're certainly willing for the psi4 recipe to do whatever works well with the c-f stack.
+#   However, at the moment this aspect of the stack seems a little fractured, so the recipe is defaulting to the
+#   mkl + intel-openmp + stashed libiomp5md.lib procedure that has worked compatibly with c-f for the last couple years.
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - make                                   # [not win]
+    - ninja                                  # [win]
+    - llvm-openmp                            # [not win]
+  host:
+    - eigen
+    - boost-cpp
+    - mkl-devel                              # [not win]
+    - mkl-devel {{ mkl }}                    # [win]
+    - mkl                                    # [not win]
+    - mkl {{ mkl }}                          # [win]
+    - pybind11
+    - pybind11-abi
+    - python
+    # qc req'd
+    - gau2grid
+    - libint {{ libint }}
+    - libxc-c
+    - optking
+    - qcelemental
+    - qcengine
+    # qc opt'l, build-time
+    - dkh
+  run:
+    - llvm-openmp                            # [not win]
+    - intel-openmp                           # [win]
+    - libblas * *mkl
+    - mkl                                    # [not win]
+    - mkl {{ mkl }}                          # [win]
+    - msgpack-python
+    - networkx
+    - numpy
+    - python
+    - scipy
+    # qc
+    - libint {{ libint }}
+    - {{ pin_compatible('optking', upper_bound='0.5') }}
+    - {{ pin_compatible('qcelemental', upper_bound='0.50') }}
+    - {{ pin_compatible('qcengine', upper_bound='0.50') }}
```

Comment:
There are static linking / overdepending warnings in the logs for these libraries. Can you explain why they are needed at both runtime and build time if they aren't dynamically linked? 

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=690074&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=13902

Reply:
These three are pure-python packages with no `run_exports`. Their important role is in `requirements/run`, and the upper bound is because there's some API-breaking changes planned in the next year, and I wanted to isolate the built packages from those. Because these three are required dependencies, if they're not present at cmake configure-time, psi4 will download and vendor them, so their presence in `requirements/host` avoids that.

Reply:
I think you're going to run into unresolvable dependencies in the future if you do it this way because eventually the host environment will move to `optking =0.6.0` and then the run environment pinning expression will become `optking >=0.6.0,<=0.5`?

I think you want something like this:
```yaml
host:
  - optking <=0.5
run:
  - {{ pin_compatible('optking', max_pin='x.x') }}
```
Such that opking is always <=0.5 and the same minor version in the host and run environments.

Reply:
Thanks for the analysis. Agreed that the original setup will take a little management of host v run to avoid the situation you describe. optking is fairly new (Dec 2022) and has a much faster release cycle (whenever) than psi4 (twice yearly), so I wanted to leave room for optking bugfixes that psi4 could access (so not pinning to x.x) and also put an upper bound so if optking finds it needs to do a breaking change, this version of psi can be insulated. It's some management, but I'm pretty sure an incompatibility will show up at build-time for me, not at solve-time for user, so I'm ok with it if you are.

Reply:
I moved the upper bound for optking out to v0.15, so all three have 12+ versions of space before hitting the limit.

Reply:
I mean this comment to apply to all three of optking, qcelemental, and qcengine. Manually specifying upper_bound is just a very unusual use of these pinning expressions.

> I wanted to leave room for optking bugfixes that psi4 could access (so not pinning to x.x)

Pinning to x.x does allow for bugfixes? For example:

```
host:
  - optking 0.5.2
run:
  - {{ pin_compatible('optking', max_pin='x.x') }}
  # This template would be replaced with optking >=0.5.2,<0.6.0
```
Are the bugfixes for optking not shipped via increment of the patch (major.minor.patch) version number?

Reply:
To allow for the widest compatability (accept bugfixes) whilst protecting against API breaks, you should use max_pin='x.x' for a package whose major version is zero. You can always rebuilt psi4 when new versions of these packages become available.

Reply:
I should have said "room for bugfixes and development". optking is new-ish and likely to be fast moving, so I was just being cautious and leaving room for an API breaking change less momentous than 1.0. I can release the constraint if you like.

The qcelemental and qcengine are in a stack with qcfractal (presently at v0.15.8 on c-f). But the last has an imminent API breaking change at 0.50 (currently in development off-c-f at https://anaconda.org/qcarchive/qcfractal/files), and once that's ready, qcelemental and qcengine will have a similar scheduled API break that we'll align at v0.50 for consistency. So I think those two are important to keep.

fwiw, I've used the upper_bound pattern before, and it's worked cleanly for the users (all their complaints come to me :-)  .